AutoRouteManager.php

```diff
@@ -77,7 +104,7 @@ public function buildUriContextCollection(UriContextCollection $uriContextCollec
 
             if (!$autoRoute) {
                 $autoRouteTag = $this->adapter->generateAutoRouteTag($uriContext);
-                $autoRoute = $this->adapter->createAutoRoute($uriContext, $uriContext->getSubjectObject(), $autoRouteTag);
+                $autoRoute = $this->adapter->createAutoRoute($uriContext, $uriContextCollection->getSubjectObject(), $autoRouteTag);
```

Comment:
Perhaps add comment here to explain why we are passing the collection's subject object (because the `UriContext` might contain a translated instance).